### PR TITLE
PMM-6395 VMagent integration

### PR DESCRIPTION
**Pull request by f41gh7**
https://github.com/Percona-Lab/pmm-submodules/pull/1185
https://github.com/percona/pmm/pull/603
https://github.com/percona/pmm-admin/pull/122
https://github.com/percona/pmm-agent/pull/201
https://github.com/percona/pmm-managed/pull/501
https://github.com/percona/grafana-dashboards/pull/1034
https://github.com/Percona-Lab/pmm-api-tests/pull/140

### CODE REVIEW

File: commands/inventory/add_agent_mongodb_exporter.go [hunk 1]
```diff
@@ -101,4 +103,5 @@ func init() {
 	AddAgentMongodbExporterC.Flag("skip-connection-check", "Skip connection check").BoolVar(&AddAgentMongodbExporter.SkipConnectionCheck)
 	AddAgentMongodbExporterC.Flag("tls", "Use TLS to connect to the database").BoolVar(&AddAgentMongodbExporter.TLS)
 	AddAgentMongodbExporterC.Flag("tls-skip-verify", "Skip TLS certificates validation").BoolVar(&AddAgentMongodbExporter.TLSSkipVerify)
+	AddAgentMongodbExporterC.Flag("use_vmagent", "When false, pull model is used. By default, its true").Default("true").BoolVar(&AddAgentMongodbExporter.UseVMAgent)
```
Comment (BupycHuk) [inline commands/inventory/add_agent_mongodb_exporter.go]: Have you discussed about usage of this flag?

Comment (f41gh7) [reply commands/inventory/add_agent_mongodb_exporter.go]: @AlekSi  suggested to add `use_vmagent` to inventory api, with it user can disable push model for some exporters. 

By default, push model detects at the server by matching client version for management api and by reading this flag for inventory api.

Comment (BupycHuk) [reply commands/inventory/add_agent_mongodb_exporter.go]: does this mean that vm-agent doesn't support pull model?

Comment (BupycHuk) [inline commands/inventory/add_agent_mongodb_exporter.go]: The second problem that I see here, it's related to true by default https://github.com/alecthomas/kingpin/issues/243.
There is nothing in help how to disable it.

Comment (f41gh7) [reply commands/inventory/add_agent_mongodb_exporter.go]: old code

Comment (f41gh7) [reply commands/inventory/add_agent_mongodb_exporter.go]: old code

File: commands/management/add_mongodb.go [hunk 2]
```diff
@@ -20,6 +20,8 @@ import (
 	"os"
 	"strings"
 
+	"github.com/AlekSi/pointer"
+
```
Comment (BupycHuk) [inline commands/management/add_mongodb.go]: let's remove this blank line

Comment (BupycHuk) [reply commands/management/add_mongodb.go]: and in other places, please.
Only 3 groups in imports
1. go packages
2. packages from other repositories
3. packages from the current repo.

File: commands/management/management.go [hunk 3]
```diff
@@ -37,4 +37,6 @@ var (
 	}
 
 	allServiceTypesKeys = []string{"mysql", "mongodb", "postgresql", "proxysql", "external"}
+
+	metricsModes = []string{"AUTO", "PUSH", "PULL"}
```
Comment (BupycHuk) [inline commands/management/management.go]: let's use lowercase

File: commands/config.go [hunk 4]
```diff
@@ -148,4 +153,6 @@ func init() {
 	ConfigC.Flag("az", "Node availability zone").StringVar(&Config.Az)
 
 	ConfigC.Flag("force", "Remove Node with that name with all dependent Services and Agents if one exist").BoolVar(&Config.Force)
+	ConfigC.Flag("metrics-mode", "Metrics flow mode for agents node-exporter, can be push - agent will push metrics,"+
+		" pull - server scrape metrics from agent  or auto - chosen by server.").EnumVar(&Config.MetricsMode, "AUTO", "PUSH", "PULL")
```
Comment (BupycHuk) [inline commands/config.go]: let's use lowercase

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: commands/config.go (hunk 4 above)
Comment (BupycHuk) [reply commands/config.go]: we should set `auto` as default value

Comment (BupycHuk) [reply commands/config.go]: the same in management commands

File: commands/management/management.go (hunk 3 above)
Comment (f41gh7) [reply commands/management/management.go]: updated

File: commands/management/add_mongodb.go (hunk 2 above)
Comment (f41gh7) [reply commands/management/add_mongodb.go]: thanks, will watch for it.

File: commands/config.go (hunk 4 above)
Comment (f41gh7) [reply commands/config.go]: done

**Review by BupycHuk — APPROVED** [review]

**Review by BupycHuk — CHANGES_REQUESTED** [review]
We forgot about proxysql

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**